scripts/validatePackageFiles.ts

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+const fs = require('fs');
+const path = require('path');
+const MissingRequiredFile = require('./errors/MissingRequiredFile');
+const FoundNonRequiredFile = require('./errors/FoundNonRequiredFile');
+
+interface IFileStructureLevel{
```

Comment:
Maybe i don't quite understand bu:
1. I'd move this to a file of its own, like FileStructure, FileStructureLevel
2. How does a file structure tree look like?
3. Isn't there a representation of folders and files tree, that we can use instead of 'reinventing the wheel'?

Reply:
also, is this script used only for tests? if so, please move to the tests folder

Reply:
A few things: 

1. We can move interfaces to their own files, it's just the interfaces though, so that felt a bit strange
2. The shape is included in the interface, it's a key/value pair of anything you want, specifically, we allow a flag and the end value can be either a string or a boolean.
3. This is (almost) the same file structure representation that's used by Mock-FS (Which we use)

This script is not used only for tests, it's used to validate the output of the `createEntryPoints` script